care/facility/api/viewsets/patient_sample.py

```diff
@@ -41,23 +43,18 @@ class PatientSampleViewSet(UserAccessMixin, viewsets.ModelViewSet):
     filterset_class = PatientSampleFilterSet
     http_method_names = ["get", "post", "patch", "delete"]
 
-    def get_serializer(self, *args, **kwargs):
-        """
-        Injects the patient data to serializer
-        """
-        try:
-            kwargs["data"]["patient"] = self.kwargs["patient_pk"]
-        except KeyError:
-            # In read serializers, "data" will not be present.
-            pass
-        return super().get_serializer(*args, **kwargs)
-
     def get_serializer_class(self):
```

Comment:
```
            def get_serializer_class(self):
                if self.action == "retrieve":
                     return PatientSampleDetailSerializer
                return self.serializer_class
```
?

Reply:
There is a convention that there should be only one `return` statement. 
https://hackerchick.com/religious-war-48293-single-vs-multiple/

Reply:
Bad example to quote, the author says he prefers multiple returns over multiple if else statements :D